src/util/datablock/datablock.c

```diff
@@ -122,8 +122,12 @@ void DataBlock_AddItems(DataBlock *dataBlock, size_t itemCount, DataBlockIterato
     if(it) {
         int step = 1;
         Block *block = ACTIVE_BLOCK(dataBlock);
-        *it = DataBlockIterator_New(block, dataBlock->itemCount,
-                                    dataBlock->itemCount + itemCount, step);
+        if (*it) {
+            DataBlockIterator_Extend(*it, itemCount);
+        } else {
+            *it = DataBlockIterator_New(block, dataBlock->itemCount,
+                                        dataBlock->itemCount + itemCount, step);
+        }
```

Comment:
Previously, calls to `DataBlock_AddItems` would allocate over an iterator if one was already present. This resolves that leak, provided that `DataBlockIterator`s are set to NULL on first creation from now on.

Reply:
Can you please point out where we've passed an initialised iterator to `DataBlock_AddItems`?
As I see it calls to `DataBlock_AddItems` MUST pass an uninitialised iterator, as a result `DataBlockIterator_Extend` should be removed.

Reply:
Sorry, I've wrongly described myself, what I meant was, the intention of DataBlock_AddItems is to get an uninitialised iterator, obviously if caller passes an initialised iterator it will get overwritten,